main.py

```diff
@@ -95,17 +180,43 @@ def draw() -> Shape:
         if events.type == generate_new_round and current_round <= 24:
             current_round = current_round + 1
 
-            if current_round > 24:
+            if current_round > 44:
+                data_management.database_commit()
                 draw_end_game()
-            else:
+            if current_round < 12:
                 clickable = True
                 winning_shape = draw()
                 last_shape = winning_shape.type
+                continue
+            if 12 < current_round < 24:
+                clickable = True
+                winning_shape = draw_second_stage()
+                last_shape = winning_shape.type
+                continue
 
         if events.type == pygame.MOUSEBUTTONDOWN and current_round <= 24 and clickable is True:
             clicked = SCREEN.get_at(pygame.mouse.get_pos())
             if clicked == winning_shape.colour:
                 clickable = False
                 print(f"You clicked the {winning_shape.type} successfully.")
-
+                if current_round == 1:
```

Comment:
This isn't needed anymore if you follow my feedback on using update with the timestamps each time and removing the baseline separation.

Reply:
Ok, but can you please clarify the above comment, cause I didn't fully get it

Reply:
@pedrorribeiro You may actually need it, it's this part of `ManageTimestamps(TimeStamps):` you do not need.

```python
        if self.stage == 'Baseline':
            self.first_timestamp(self.ts_dict)
        else:
            self.timestamp_add(self.ts_dict)
```

(Both can just use `self.timestamp_add(self.ts_dict)`, there's no need for a `first_timestamp()` call.)

The question I have is if "Baseline" is _**always**_ in the allowed stages, or if it's only in the allowed stages for round 1? If it's always in the allowed stages, my original comment holds true and you can remove that check - otherwise it's needed.